generate/src/cli.js

```diff
@@ -0,0 +1,26 @@
+// Get slug as argument
+const slug = process.argv[2];
+
+// required to write files
+const fs = require('fs')
+
+// Required to run HTTP requests (usually handled by browser)
+global.XMLHttpRequest = require('xhr2');
+
+// Link to compiled Elm code main.js and run it
+var Elm = require("./main").Elm;
+var main = Elm.Main.init({ flags: slug });
```

Comment:
Instead of only passing the slug to the Elm worker, I'd suggest downloading the canonical data first in JS and passing both the slug and canonical data in the worker. This avoids both the need for the xhr2 dependency (+ package.json and package lock) and the webdata elm dependency to remove a bunch of code. You can pass the downloaded json as a `Json.Decode.Value` in the flags and decode it with the same decoder already in your elm code.

Reply:
Yes, I agree, it would be cleaner.